pkg/providers/vsphere/vmlifecycle/bootstrap_templatedata.go

```diff
@@ -45,12 +45,14 @@ func GetTemplateRenderFunc(
 		Nameservers: bsArgs.DNSServers,
 	}
 
-	// Oh dear. The VM itself really should not have been included here.
+	// Use deepCopy of the VM to prevent issues due to down-converting the actual VM on
+	// next reconcile
+	copyVM := vmCtx.VM.DeepCopy()
```

Comment:
Do we want to give each `ConvertFrom` their own copy instead? Or at least restore the annotations after the first until we fix the aliasing in the conversion code

Reply:
Using separate copies. Let me know if you have a better suggestion for naming these copies. updated

Reply:
Oh I guess I wouldn't even give them names: `ConvertFrom(vmCtx.VM.DeepCopy())`? . And leave your comment as to why we're doing that.